rhombus/private/box.rkt

```diff
@@ -33,18 +25,42 @@
   (provide box-method-table
            box-mutator-method-table))
 
-(define-for-syntax box-static-infos
-  #'((#%dot-provider box-instance)))
-
-(define-name-root Box
-  #:fields
+(define-primitive-class Box box
+  #:lift-declaration
+  #:constructor-static-info #f
+  #:existing
+  #:opaque
+  #:fields ()
+  #:namespace-fields
   (now_of
-   later_of))
+   later_of
+   )
+  #:properties
+  ;; TODO undocumented (as function)
+  ([value Box.value #:mutator Box.value
+          (lambda (e)
+            (syntax-local-static-info e #'unbox))]
+   )
+  #:methods
+  ())
 
 (define/arity (Box v)
   #:static-infos ((#%call-result #,box-static-infos))
   (box v))
 
+(define (check-box who b)
+  (unless (box? b)
+    (raise-argument-error* who rhombus-realm "Box" b)))
+
+(define/arity Box.value
```

Comment:
Same issue as `Array.get`: the original intent was to let `unbox` be used directly.